File: src/slownews/controller/SignUp.java
package slownews.controller;

import slownews.model.User;

import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.Map;

@WebServlet("/signUp")
public class SignUp extends HttpServlet {

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        User user = new User();
        user.setLogin(request.getParameter("login"));
        user.setPassword(request.getParameter("password"));
        user.setEmail(request.getParameter("email"));
        user.setNumber(request.getParameter("number"));

        Map<String, User> users = null;
        ServletContext context = request.getSession().getServletContext();
        Object obj = context.getAttribute("users");
        if (obj instanceof Map) {
            users = (Map) obj;
        }

        if (!users.containsKey(user.getLogin())) {
            users.put(user.getLogin(), user);
            request.getRequestDispatcher("jsp/signIn.jsp").forward(request, response);
        } else {
            request.setAttribute("message", "user with that name already registered");
            request.getRequestDispatcher("jsp/signUp.jsp").forward(request, response);
        }
    }

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        if (request.getSession().getAttribute("user") == null) {
            request.getRequestDispatcher("jsp/signUp.jsp").forward(request, response);
        } else {
            request.getRequestDispatcher("/").forward(request, response);
        }
    }
}

File: web/js/signUp.js
function check(login, password, email, number) {
    var login = document.getElementById(login).value;
    var password = document.getElementById(password).value;
    var email = document.getElementById(email).value;
    var number = document.getElementById(number).value;
    var message = '';

    if ((login == null) || (login == '') || (login == undefined)) {
        message = 'Enter login';
    }
    if ((password == null) || (password == '') || (password == undefined)) {
        message = message + '\n' + 'Enter password';
    }
    if ((email == null) || (email == '') || (email == undefined)) {
        message = message + '\n' + 'Enter email';
    }
    if ((number == null) || (number == '') || (number == undefined)) {
        message = message + '\n' + 'Enter number';
    }
    if (message.length == 0) {
        return true;
    }
    alert(message);
    return false;
}